### ISSUE TITLE
Evaluate Regular expression in setup_run_environment

### ISSUE THREAD

**Issue by Billae** [LLNL/spack#21907]
Hello,

This issue continued  this one #21763) which want to set additionnal path for a Paraview plugin project.
 
I have a project which is a paraview plugin and i need to set the PV_PLUGIN_PATH variable to make it available in paraview.

To detect my project as a plugin, Paraview need the full path to spack/opt/.../project/lib( or lib64)/paraview-<version of paraview>/plugin/project.

As when i install my plugin I did not know which is the paraview version associated, I need to make this path generic.
To this end I try to add in my recipe

    def setup_run_environment(self, env):
        env.prepend_path('PV_PLUGIN_PATH', join_path(self.prefix, 'lib*', '**', 'plugins','project'))

However, the regular expression usage is not detected when i run paraview which say to me "searching plugins in spack/opt/.../project/lib*/**/plugin/project.".
When i try to echo this path it works fine:

> echo $PV_PLUGIN_PATH
> spack/opt/.../project/lib64/paraview-5.9/plugin/project.

In short, I need to evaluate this regular expression during the module generation, and if possible in the setup_run_environment method. Is it possible ?

Thanks
Eloïse

**Comment by adamjstewart** [comment]
If you `import glob` and put the `join_path` inside of `glob.glob(...)[0]` it will have the effect you desire.

**Comment by Billae** [comment]
Thanks a lot ! It works just fine =)
